core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -300,36 +377,41 @@ object ConsumerGroupCommand {
     // `consumer` is only needed for `describe`, so we instantiate it lazily
     private var consumer: KafkaConsumer[String, String] = null
 
-    def list() {
-      adminClient.listAllConsumerGroupsFlattened().foreach(x => println(x.groupId))
+    def listGroups(): List[String] = {
+      adminClient.listAllConsumerGroupsFlattened().map { _.groupId }.toList
     }
 
-    protected def describeGroup(group: String) {
-      adminClient.describeConsumerGroup(group) match {
-        case None => println(s"Consumer group `${group}` does not exist.")
-        case Some(consumerSummaries) =>
-          if (consumerSummaries.isEmpty)
-            println(s"Consumer group `${group}` is rebalancing.")
-          else {
-            val consumer = getConsumer()
-            printDescribeHeader()
-            consumerSummaries.foreach { consumerSummary =>
-              val topicPartitions = consumerSummary.assignment.map(tp => TopicAndPartition(tp.topic, tp.partition))
-              val partitionOffsets = topicPartitions.flatMap { topicPartition =>
-                Option(consumer.committed(new TopicPartition(topicPartition.topic, topicPartition.partition))).map { offsetAndMetadata =>
-                  topicPartition -> offsetAndMetadata.offset
-                }
-              }.toMap
-              describeTopicPartition(group, topicPartitions, partitionOffsets.get,
-                _ => Some(s"${consumerSummary.clientId}_${consumerSummary.clientHost}"))
+    protected def collectGroupAssignment(group: String): (String, Option[Seq[PartitionAssignmentState]]) = {
+      val describConsumerGroupResult = adminClient.describeConsumerGroup(group)
+      (describConsumerGroupResult._1,
+        describConsumerGroupResult._2 match {
+          case None =>
+            None
+          case Some((coordinator, consumerSummaries)) =>
+            if (consumerSummaries.isEmpty)
+              Some(Array[PartitionAssignmentState]())
+            else {
+              val consumer = getConsumer()
+              var assignmentRows = Array[PartitionAssignmentState]()
+              consumerSummaries.foreach { consumerSummary =>
```

Comment:
Looks like another case where we might be able to change the `foreach` to a `map` with a `toArray` at the end.